hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/CellBlockBuilder.java

```diff
@@ -294,14 +296,15 @@ private ByteBuffer decompress(CompressionCodec compressor, InputStream cellBlock
     }
     Decompressor poolDecompressor = CodecPool.getDecompressor(compressor);
     CompressionInputStream cis = compressor.createInputStream(cellBlockStream, poolDecompressor);
-    ByteBufferOutputStream bbos;
     try {
       // TODO: This is ugly. The buffer will be resized on us if we guess wrong.
-      // TODO: Reuse buffers.
-      bbos = new ByteBufferOutputStream(osInitialSize);
-      IOUtils.copy(cis, bbos);
-      bbos.close();
-      return bbos.getByteBuffer();
+      if (this.decompressBuff.get() == null) {
+        this.decompressBuff.set(new ByteBufferOutputStream(osInitialSize));
+      }
+      ByteBufferOutputStream localBbos = this.decompressBuff.get();
+      localBbos.clear();
+      IOUtils.copy(cis, localBbos);
+      return localBbos.getByteBuffer();
```

Comment:
This will leak the internal ByteBuffer output? What if the upper layer cache this ByteBuffer and pass it to other threads...

Reply:
Ah.....That is really a problem. Let me think how to deal with this here.......

Reply:
It seems that there is no better solution than returning a copy of the underlying bytebuffer. But copy is not a reuse of buffers. Currently, the threadlocal is not a good idea here... (this should be reconsidered). May I close this PR and the related ticket ? @Apache9

Reply:
OK, should be closed as later or won't fix.